nameSpace.js

```diff
@@ -0,0 +1,14 @@
+function provide(newClass, constructor) {
+	var newClass = new constructor(document.getElementById('screen'));
```

Comment:
When you invoke *new* you are creating instance, **not class**.  And with provide you do not return anything, you just need to keep it somewhere internally. Look at this problem as put & get. With provide you are putting a class, with require you are getting it.